From 1c7130844b241c1cc2d6c97545dcf895b310eec5 Mon Sep 17 00:00:00 2001
From: Jeroen van Meeuwen (Fedora Unity) <kanarip@fedoraunity.org>
Date: Wed, 25 Nov 2009 13:42:59 +0100
Subject: [PATCH 2/2] Correct rundir permissions

---
 lib/puppet/defaults.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lib/puppet/defaults.rb b/lib/puppet/defaults.rb
index 9d992dd..7967877 100644
--- a/lib/puppet/defaults.rb
+++ b/lib/puppet/defaults.rb
@@ -62,7 +62,7 @@ module Puppet
         },
         :rundir => {
             :default => rundir,
-            :mode => 01777,
+            :mode => 0755,
             :desc => "Where Puppet PID files are kept."
         },
         :genconfig => [false,
-- 
1.6.5.2

